app/build.gradle

```diff
@@ -163,7 +163,7 @@ dependencies {
 
     androidTestImplementation "com.google.dagger:dagger:$dagger_version"
     androidTestAnnotationProcessor "com.google.dagger:dagger-compiler:$dagger_version"
-
+    implementation 'com.github.Gkemon:Android-XML-to-PDF-Generator:2.6.8'
```

Comment:
First, thanks for looking into this! 

I really like the simplicity of this library. I'm a bit worried though that the last changes in the repository are from two years ago and that there seems to be some stuff with permissions going on in the issues for different versions of Android. 

Did you do any testing with different versions? I just only did some quick tests with Android 15, but we have to support back to API level 26 (Android 8).

Reply:
I tested on Android 14, 12, and 10
I do not expect issues on Android 8, the library works down to SDK21
With automated test I have no experience. I don't do this for any of my 21 apps on F-Droid :-)

Reply:
I am also not happy with the fact that the library does not seem to be maintained at the moment.
But on the other hand this library gives us a very simple implementation for this feature, and if necessary it can be replaced by a better implementation sometime in future.

Reply:
The permissions issue does not seem to be a problem for our use case. It seems to be an issue when using Kotlin, for whatever reason. The relevant Android changes for read/write permissions should be covered with my tests on Android 10, 12, 14

Reply:
Yes, I agree that it's at least easy to replace/remove in the worst case.

I will merge the changes and see if an automated test is feasible, would be nice to have, but sometimes Android testing is quite unsatisfying 😅